Modules/_posixsubprocess.c

```diff
@@ -561,26 +583,32 @@ subprocess_fork_exec(PyObject* self, PyObject *args)
     PyObject *env_list, *preexec_fn;
     PyObject *process_args, *converted_args = NULL, *fast_args = NULL;
     PyObject *preexec_fn_args_tuple = NULL;
+    PyObject *groups_list;
     int p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite;
     int errpipe_read, errpipe_write, close_fds, restore_signals;
     int call_setsid;
+    int call_setgid, call_setuid;
+    uid_t uid;
+    gid_t groups[MAX_GROUPS];
```

Comment:
On Linux, `MAX_GROUPS` is 65536. Are you sure you want to allocate 256 KB on stack?

Reply:
Would something like this be better?
```c
    gid_t *groups = NULL;
    ...
        groups = PyMem_RawCalloc(num_groups, sizeof(gid_t));
    ...
cleanup:
    ...
    if (groups)
        PyMem_RawFree(groups);
```
I would still maintain a check for `if (num_groups > MAX_GROUPS)` to make sure we never get more than MAX_GROUPS